### PR TITLE
refactor code from run_tests.py into Python module

### PR DESCRIPTION
**Pull request by dirk-thomas**

### CODE REVIEW

File: python/catkin/test_results.py [hunk 1]
```diff
@@ -31,8 +31,68 @@
...
+
+
+def ensure_junit_result_exist(filename):
+    if os.path.exists(filename):
+        # if result file exists ensure that it contains valid xml
+        tree = None
+        try:
+            tree = ElementTree(None, filename)
+        except ParseError:
+            # print('Invalid XML in result file "%s"' % filename)
```
Comment (wjwwood) [inline python/catkin/test_results.py:59]: Should this be removed?

**Comment by wjwwood** [comment]
+1

**Comment by tfoote** [comment]
+1

File: python/catkin/test_results.py (hunk 1 above)
Comment (dirk-thomas) [inline python/catkin/test_results.py:59]: Good point. But sometimes I use those commented print statements to trace issues so I will keep it for now.

**Pull request merged**